source/vir/src/well_formed.rs

```diff
@@ -828,25 +826,28 @@ fn check_function(
         }
     }
 
-    if function.x.publish.is_some() && function.x.mode != Mode::Spec {
-        return Err(error(
-            &function.span,
-            "function is marked `open` but it is not a `spec` function",
-        ));
-    }
-
     if function.x.is_main() && function.x.mode != Mode::Exec {
         return Err(error(&function.span, "`main` function should be #[verifier::exec]"));
     }
 
-    if function.x.publish.is_some()
-        && function.x.visibility.is_private_to(&function.x.owning_module)
-    {
+    // These should be true by construction in Rust->VIR; sanity check them here.
+    if !function.x.body_visibility.at_least_as_restrictive_as(&function.x.visibility) {
```

Comment:
Should this be a panic instead? (Given that this should be "true by construction")

Reply:
Well an error is always better than a panic, but maybe it should say "Verus internal error" or something.

Reply:
Agreed. Don't we have something like that? (maybe we should make a variant of one of the macros in https://github.com/verus-lang/verus/blob/main/source/rust_verify/src/util.rs for this purpose.)

Reply:
This may be helpful: https://github.com/verus-lang/verus/pull/1453
If it looks good, we can replace it after merging.